frontend/src/stores/user.ts

```diff
@@ -1,167 +1,165 @@
-                await this.logout();
-            }
-        },
-        async updateUserProfile(payload: UpdateMeDTO) {
-            const mainStore = useMainStore();
-            const loadingNotification = {content: 'saving', showProgress: true};
-            try {
-                mainStore.addNotification(loadingNotification);
-                this.userProfile = await api.updateMe(payload);
-                mainStore.removeNotification(loadingNotification);
-                mainStore.addNotification({content: 'Profile successfully updated', color: TYPE.SUCCESS});
-            } catch (error) {
-                await mainStore.checkApiError(error);
-                throw new Error(error.response.data.detail);
-            }
-        },
-        async checkLoggedIn() {
-            const mainStore = useMainStore();
-            if (mainStore.authAvailable) {
-                if (!this.isLoggedIn) {
-                    let token = this.token;
-                    if (!token) {
-                        const localToken = getLocalToken();
-                        if (localToken) {
-                            this.token = localToken;
-                            token = localToken;
-                        }
-                    }
-                    if (token) {
-                        const response = await api.getUserAndAppSettings();
-                        this.isLoggedIn = true;
-                        this.userProfile = response.user;
-                        let appConfig = response.app;
-                        mainStore.setAppSettings(appConfig);
-                    } else {
-                        this.removeLogin();
-                    }
-                }
-            } else {
-                const response = await api.getUserAndAppSettings();
-                this.isLoggedIn = true;
-                this.userProfile = response.user;
-                let appConfig = response.app;
-                mainStore.setAppSettings(appConfig);
+    async updateUserProfile(payload: UpdateMeDTO) {
+      const mainStore = useMainStore();
+      const loadingNotification = { content: 'saving', showProgress: true };
+      try {
+        mainStore.addNotification(loadingNotification);
+        this.userProfile = await api.updateMe(payload);
+        mainStore.removeNotification(loadingNotification);
+        mainStore.addNotification({ content: 'Profile successfully updated', color: TYPE.SUCCESS });
+      } catch (error) {
+        await mainStore.checkApiError(error);
+        throw new Error(error.response.data.detail);
+      }
+    },
+    async checkLoggedIn() {
+      const mainStore = useMainStore();
+      if (mainStore.authAvailable) {
+        if (!this.isLoggedIn) {
+          let token = this.token;
+          if (!token) {
+            const localToken = getLocalToken();
+            if (localToken) {
+              this.token = localToken;
+              token = localToken;
```

Comment:
I don't think that we really need a `token` variable since it's always equals to `this.token` 

It seems (need to verify) that we can simplify everything by just

```
this.token = this.token || getLocalToken()
```

Reply:
this doesn't work because `getLocalToken()` may output null

Reply:
In that case something like this should work: `this.token = this.token || getLocalToken() || ''`